src/software/simulation/physics/physics_robot.h

```diff
@@ -39,9 +40,11 @@ class PhysicsRobot
      * @param world A shared_ptr to a Box2D World
      * @param robot_state The initial robot state
      * @param mass_kg The mass of the robot in kg
+     * @param post_world_update_functions A buffer of functions to be executed
+     * after each world time step
      */
     explicit PhysicsRobot(const RobotId id, std::shared_ptr<b2World> world,
-                          const RobotState &robot_state, const double mass_kg);
+                          const RobotState &robot_state, const double mass_kg, std::shared_ptr<ThreadSafeBuffer<std::function<void()>>> post_world_update_functions);
```

Comment:
Doc'ing offline discussion here:

Instead of passing in a buffer to push to like this, we've decided to make a `std::queue<std::function<void()>>` that is stored internally in the `PhysicsRobot` that represents the op's to perform after the physics. Then we can predeclare `PhysicsWorld` in `physics_robot.h`, and make a protected `PhysicsRobot::runPostPhysicsTick` function that the `PhysicsWorld` can call for all it's owned `PhysicRobot`'s.

Reply:
done